### PR TITLE
Availability check based on providers-common gem

### PR DESCRIPTION
**Pull request by slemrmartin**
**Issue** https://github.com/RedHatInsights/sources-api/issues/206

This PR moves Source#availability_check from AnsibleTower to providers-common. 

---

**Depends on**:

* [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/25
* [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/28
* [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/33
* [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/34
---

[TPINVTRY-944](https://projects.engineering.redhat.com/browse/TPINVTRY-944)

### CODE REVIEW

**Comment by syncrou** [comment]
@slemrmartin was hoping to merge this ( and the others ) but I can't get Travis to pass.  It seemed like it was a Travis issue, ( missing file reference ) but Travis failed twice https://travis-ci.org/github/RedHatInsights/topological_inventory-azure/builds/699472110?utm_source=github_status&utm_medium=notification

**Comment by gmcculloug** [comment]
@slemrmartin @syncrou The `spec` folder in https://github.com/RedHatInsights/topological_inventory-providers-common is not part of the gem release.  The `spec/support/shared/availability_check.rb` file was added to that repo 8 days ago.

Here is a file listing of the gem I have locally:
```
 ~/.gem/ruby/2.6.6/gems/topological_inventory-providers-common-1.0.4 $ ls -l 
total 48
-rw-r--r--  1 gmccullough  staff  1390 Jun 17 17:58 CHANGELOG.md
-rw-r--r--  1 gmccullough  staff   339 Jun 17 17:58 Gemfile
-rw-r--r--  1 gmccullough  staff  1079 Jun 17 17:58 LICENSE.txt
-rw-r--r--  1 gmccullough  staff   890 Jun 17 17:58 README.md
-rw-r--r--  1 gmccullough  staff   117 Jun 17 17:58 Rakefile
drwxr-xr-x  4 gmccullough  staff   128 Jun 17 17:58 bin
drwxr-xr-x  3 gmccullough  staff    96 Jun 17 17:58 lib
-rw-r--r--  1 gmccullough  staff  2035 Jun 17 17:58 topological_inventory-providers-common.gemspec
```

**Comment by slemrmartin** [comment]
oh, that's an issue! I haven't found any article about at. 

@syncrou What do you suggest? 
I see 2 possibilities

1. Move `/spec/support/` to `/support/` (I hope it will be released) or 
2. Copy-paste the spec to all repos

**Comment by slemrmartin** [comment]
or this one can help: https://github.com/RedHatInsights/topological_inventory-providers-common/pull/33

File: Gemfile [hunk 1]
```diff
@@ -13,9 +13,9 @@ gem "optimist"
 gem "prometheus_exporter", "~> 0.4.5"
 gem "rake", ">= 12.3.3"
 gem "rest-client", "~>2.0"
-gem "sources-api-client", "~> 1.0"
+gem "sources-api-client", "~> 3.0"
 gem "topological_inventory-ingress_api-client", "~> 1.0.2"
-gem "topological_inventory-providers-common", "~> 1.0.2"
+gem "topological_inventory-providers-common", "~> 1.0.5"
```
Comment (syncrou) [inline Gemfile:18]: Can we change this to `~> 1.0`?

Comment (slemrmartin) [reply Gemfile:18]: we can change it only to `'~> 1.0', '>= 1.0.5'` so it doesn't worth

**Review by syncrou — APPROVED** [review]
LGTM 👍  - We probably have to discuss the gem versioning, though it may be that Hakiri is holding us to such specific requirements. Those changes ( if any ) can be in a follow up.

**Pull request merged**